src/plugins/intel_npu/src/common/src/igraph.cpp

```diff
@@ -17,12 +17,12 @@ namespace intel_npu {
 IGraph::IGraph(ze_graph_handle_t handle,
                NetworkMetadata metadata,
                const Config& config,
-               std::optional<std::vector<uint8_t>> blob)
+               std::optional<std::unique_ptr<BlobContainer>> blobPtr)
```

Comment:
not sure that we still need std::optional now

Reply:
For `compile_model` path, should we create an empty `BlobContainer` class instead and drop `std::optional`? But this will require `BlobContainer` class to be instantiable, so no pure virtual functions as mentioned above.

Reply:
can we just use nullptr instead of creating an empty blobContainer?